pkg/vault/vault.go

```diff
@@ -17,6 +17,44 @@ func NewVaultToken(vc *vaultClient) (string, error) {
 	return NewAuthClient(vc.config).GetVaultToken(vc)
 }
 
+// ListEngines fills a map with the secrets engines pulled from Vault.
+func ListEngines(ctx context.Context, path string) ([]string, error) {
+	environment := os.Getenv("ENVIRONMENT")
+
+	if environment == "production" {
+		log.SetFormatter(&log.JSONFormatter{})
+		log.SetLevel(log.WarnLevel)
+	} else {
+		log.SetFormatter(&log.TextFormatter{})
+		log.SetLevel(log.TraceLevel)
+	}
+
+	config, err := loadVaultEnvironment()
+	if err != nil {
+		return nil, fmt.Errorf("load client environment: %v", err)
+	}
+
+	vc, err := NewVaultClient(ctx, config)
+
+	if err != nil {
+		return nil, fmt.Errorf("error initializing vault client: %v", err)
+	}
+
+	if config.traceEnabled {
+		var span *trace.Span
+		ctx, span = trace.StartSpan(ctx, fmt.Sprintf("%s/ListEngines", config.tracePrefix))
+		defer span.End()
+	}
```

Comment:
with the tracer helper these lines will DRY up as well

```suggestion
	 vc.tracer.trace(fmt.Sprintf("%s/ListEngines", vc.config.tracePrefix))
```